File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s: str):
    if len(s) % 2 == 1:
        return 0
    
    stack = []
    for char in s:
        if not(stack) or stack[-1] != char:
            stack.append(char)
        else: # stack[-1] == char:
            stack.pop()
    
    return 1 if not(stack) else 0

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.Stack;

class Solution {
    public int solution(String s) {
        if (s.length() % 2 == 1) {
            return 0;
        }

        Stack<Character> stack = new Stack<>();
        for (char c : s.toCharArray()) {
            if (stack.isEmpty() || stack.peek() != c) {
                stack.push(c);
            } else {
                stack.pop();
            }
        }

        return stack.isEmpty() ? 1 : 0;
    }
}